Move to addon v2

ctjhoa opened the pull request:
~~This addon is already in ember-auto-import v2 so it should not be a breaking change.~~
:warning:  This is breaking change
Yes this addon is already ember-auto-import v2 but ember app must be `ember-auto-import@2.4.3`  minimum to fix the `@glimmer/validator` issue (cf https://github.com/emberjs/ember-render-modifiers/pull/66#issuecomment-1277354436)

This is based on addon v2 with yarn monorepo setup.
It follows
- https://github.com/embroider-build/addon-blueprint
- https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md

ctjhoa committed:
Relax ember-source peer dependency (#61)

SergeAstapov commented on packages/ember-render-modifiers/dist/modifiers/did-insert.js:
```diff
@@ -0,0 +1,63 @@
+import { setModifierManager, capabilities } from '@ember/modifier';
```
@ctjhoa dist folder should be ignore and should be be checked in to git

ctjhoa replied:
fixed

SergeAstapov commented on package.json:
```diff
@@ -1,95 +1,49 @@
-  "ember-addon": {
-    "configPath": "tests/dummy/config"
+  "resolutions": {
+    "**/@glimmer/validator": "0.65.4"
```
this is fishy, I don't think this is needed

ctjhoa replied:
I agree, this was an attempt to fix `@glimmer/validator` issue. cf https://github.com/emberjs/ember-render-modifiers/pull/66#issuecomment-1277354436

ctjhoa committed:
Move to addon v2

ctjhoa commented:
I'm stuck with this PR.
There is an issue with `@glimmer/validator` which I'll try to sum up.

Ember embed it's own `@glimmer/validator` version as [devDependencies](https://github.com/emberjs/ember.js/blob/v4.9.0-alpha.6/package.json#L98). 
So `ember-source` as a package doesn't have an explicit dependency on `@glimmer/validator`.

On the other side, since https://github.com/emberjs/ember-render-modifiers/pull/54, `ember-render-modifier` is dependent on  `@glimmer/validator`.

In classic (non-embroider) build, `@glimmer/validator` is somehow resolved as the Ember internal dependency.

Embroider, on the other hand, relies on explicit npm dependency and expects `@glimmer/validator` to be explicity provided by `ember-source`.
This result in `@glimmer/validator` loaded twice

![2022-10-11-144941_1916x696_scrot](https://user-images.githubusercontent.com/1716173/195565872-aa42bb1e-c045-482d-9eee-5694aee13f3b.png)


Here is a list of things I've tried and do not work at the moment:
- Declare `@glimmer/validator` as optional peerDependencies
- Use yarn `resolutions` to  force only 1 version of `@glimmer/validator` 
- Tweak `@embroider/shared-internals` `ember-standard-modules.ts` to add `emberVirtualPackages.add('@glimmer/validator')`

ctjhoa commented:
https://github.com/ef4/ember-auto-import/pull/541 might solve the issue

ctjhoa committed:
Flag "@glimmer/validator" as external resolution

Read the full issue here
https://github.com/emberjs/ember-render-modifiers/pull/66#issuecomment-1277354436

ctjhoa commented:
> https://github.com/ef4/ember-auto-import/pull/541 might solve the issue
I confirm this is working. I'm waiting for a new release of `ember-auto-import`

SergeAstapov commented:
@ctjhoa minor, I think we don't need `packages/ember-render-modifiers/config/environment.js` file at all

ctjhoa committed:
Upgrade release-it and plugins

ctjhoa committed:
fixup! Move to addon v2

SergeAstapov commented on package.json:
```diff
@@ -1,95 +1,46 @@
+    "release": "release-it"
+  },
+  "volta": {
+    "node": "16.14.2",
```
@ctjhoa I think this should stay as 12 as minimum supported version and the same what was used in CI.

Or we could remove `engines.node` from `@ember/render-modifiers` as it's purely frontend code and has nothing to do with Node.js versions anymore (fully delegated to build pipeline)

ctjhoa replied:
I've removed `engines.node`
https://github.com/emberjs/ember-render-modifiers/pull/66/commits/226ff161bb72a1c2ce072fe5bc720d6500f5a2c0

As this PR introduces a breaking change, go for the latest Node LTS seems fine to me.

ctjhoa committed:
Remove "engines" section from package.json

ctjhoa committed:
Bump actions/checkout@v3

ctjhoa committed:
Bump volta-cli/action@v4

ctjhoa committed:
Emphasis ember-auto-import version

NullVoxPopuli commented:
@SergeAstapov time for re-review?

jrjohnson commented:
@ctjhoa - are you still able to work on this? I think we should be able to get it over the finish line now that we've dropped support for node < 18 and Ember < 4.12.

ctjhoa commented:
> @ctjhoa - are you still able to work on this? I think we should be able to get it over the finish line now that we've dropped support for node < 18 and Ember < 4.12.

Nope sorry I don't work on ember stack anymore and I've opened this PR 2 years ago and I didn't follow ember newest changes 🤷

SergeAstapov commented:
> @SergeAstapov time for re-review?

I guess a rebase is needed first. Or a new PR forked from this branch

jrjohnson commented:
I thought I hadn't seen your name on anything for a bit @ctjhoa, thanks for the quick response. I'll take over this work and restart it either with a rebase or a fresh start. Will probably do a pnpm migration first though.

jrjohnson commented:
I'm still on the hook for this. It's on hold as I'm not having any luck finding anyone who [can release this addon](https://github.com/emberjs/ember-render-modifiers/pull/80#issuecomment-2591177136). Want to do a release of current `master` and then get some changes in.

RobbieTheWagner commented:
@jrjohnson you can probably proceed with this now. Good luck and thank you for your help!

jrjohnson commented:
I've restarted this work in #87, but I have some questions about typescript and release it. If anyone on this thread can help with that please take a look over there.

NullVoxPopuli commented:
Thank you for your efforts! This all finally landed in https://github.com/emberjs/ember-render-modifiers/pull/89
